public/static/docs/api-reference/open.md

```diff
@@ -0,0 +1,65 @@
+# dvc.api.open()
+
+Opens a file <abbr>artifact</abbr> as a
+[file object](https://docs.python.org/3.7/glossary.html#term-file-object). May
+only be used as
+[context manager](https://www.python.org/dev/peps/pep-0343/#context-managers-in-the-standard-library).
+
+> This has similar uses as the `dvc get` and `dvc import` CLI commands.
```

Comment:
OK @Suor @shcheklein  I actually probably need help answering this one: Does this API function somehow [stream](https://docs.python.org/3.7/library/io.html) files from the source project/remote? (I've heard this somewhere.)

If so, please point me to the relevant source code in our code repo and/or documentation references so I can describe the benefits clearly here and in other docs.

Reply:
I spoke about this on Slack with Ramón and @efiop and Ruslan said
> it does stream for some remotes (not sure all of them)
> need to double check though, it was a while ago
Would be great if you can check or point me in the right direction, thanks!

Reply:
@jorgeorpinel I think we did by mentioning that all remotes that have `open` support streaming :) But here is a complete list: SSH, Azure, Google Storage, HDFS, Local, OSS, S3.

Reply:
Thanks!

Reply:
Added list in 9b58c3f8.